lib/pbench/test/unit/server/query_apis/test_index_mappings.py

```diff
@@ -30,7 +30,7 @@ def test_run_template_query(self, client, server_config, find_template):
         response body.
         """
         with client:
-            response = client.get(f"{server_config.rest_uri}/index/mappings/run")
+            response = client.get(f"{server_config.rest_uri}/index/mappings/search")
```

Comment:
The more I see the pattern, the more I think that the `index` in this URI has become meaningless. It's really too much of an Elasticsearch term. We're more ... I don't know ... getting the mappings for "views" of a dataset, whether the run summary, iteration data, TOC, or the more detailed timeseries data.

A related observation ... we currently tie "search" to the run index, but there's no real reason for this except that's the way the old dashboard search page worked. Given the flexible mappings/search mechanism we've got in the server a client could just as easily search any "view" of the data.

In the particular context of this PR, I think that using the keyword "run" instead of "search" for the current "mapping" of the dashboard search function probably makes sense, even if we don't emphasize the fact that you could just as easily use any of the keywords. 😄

Reply:
So @dbutenhof, should (perhaps not in this PR) "index" be replaced with "dataset"?  That might enable a URI of `/dataset/mappings/<keyword>`....

Reply:
I hadn't made that observation with any particular name in mind, but I like that idea a lot. The pair of `/api/v1/datasets/mappings/<type>` and `/api/v1/datasets/search/<type>` would seem very clear.

Reply:
I concur `/dataset/mappings/<keyword>` looks better, I will create an issue to track for replacing `index` with `dataset`